src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -44,28 +53,100 @@ public IEnumerable<DataSource> ListAllDataSources()
-        this._dataSourceNameToDataSource.Add(this._defaultDataSourceName, this.DataSource);
+        _defaultDataSourceName = Guid.NewGuid().ToString();
 
-        this._entityNameToDataSourceName = new Dictionary<string, string>();
+        // we will set them up with default values
+        _dataSourceNameToDataSource = new Dictionary<string, DataSource>
+        {
+            { _defaultDataSourceName, this.DataSource }
+        };
+
+        _entityNameToDataSourceName = new Dictionary<string, string>();
         foreach (KeyValuePair<string, Entity> entity in Entities)
         {
-            _entityNameToDataSourceName.TryAdd(entity.Key, this._defaultDataSourceName);
+            _entityNameToDataSourceName.TryAdd(entity.Key, _defaultDataSourceName);
         }
 
+        // Process data source and entities information for each database in multiple database scenario.
+        this.DataSourceFiles = DataSourceFiles;
+
+        if (DataSourceFiles is not null && DataSourceFiles.SourceFiles is not null)
+        {
+            IEnumerable<KeyValuePair<string, Entity>> allEntities = Entities.AsEnumerable();
+            // Iterate through all the datasource files and load the config.
+            IFileSystem fileSystem = new FileSystem();
+            FileSystemRuntimeConfigLoader loader = new(fileSystem);
+
+            foreach (string dataSourceFile in DataSourceFiles.SourceFiles)
+            {
+                if (loader.TryLoadConfig(dataSourceFile, out RuntimeConfig? config))
+                {
+                    try
+                    {
+                        _dataSourceNameToDataSource = _dataSourceNameToDataSource.Concat(config._dataSourceNameToDataSource).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        _entityNameToDataSourceName = _entityNameToDataSourceName.Concat(config._entityNameToDataSourceName).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        allEntities = allEntities.Concat(config.Entities.AsEnumerable());
+                    }
+                    catch (Exception e)
+                    {
+                        // Errors could include duplicate datasource names, duplicate entity names, etc.
+                        throw new DataApiBuilderException(
+                            $"Error while loading datasource file {dataSourceFile} with exception {e.Message}",
+                            HttpStatusCode.BadRequest,
+                            DataApiBuilderException.SubStatusCodes.ConfigValidationError,
+                            e.InnerException);
+                    }
+                }
+            }
+
+            this.Entities = new RuntimeEntities(allEntities.ToDictionary(x => x.Key, x => x.Value));
+        }
+
+        SqlDataSourceUsed = _dataSourceNameToDataSource.Values.Any
+            (x => x.DatabaseType == DatabaseType.MSSQL || x.DatabaseType == DatabaseType.PostgreSQL || x.DatabaseType == DatabaseType.MySQL);
```

Comment:
if we are adding code to check for CosmosDB_PostgreSQL, it should be added here.

Reply:
I had skipped cosmosDB_PostGresSQL all together, was we do not have support for it as of today so dont want to add that as part of this pr.